matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java

```diff
@@ -481,15 +488,25 @@ private void setAccess()
     }
 
     /**
-     * Checks if current user is owner of one patient.
+     * Checks if current user or one of his groups is owner of one patient.
      */
     private boolean isUserOwner()
     {
-        User user = USER_MANAGER.getCurrentUser();
-        DocumentReference userRef = user.getProfileDocument();
-        if (this.patient != null && userRef.equals(ACCESS_HELPER.getOwner(this.patient).getUser())) {
+        if (this.patient == null) {
+            return false;
+        }
+        User currentUser = USER_MANAGER.getCurrentUser();
+        DocumentReference userRef = currentUser.getProfileDocument();
+        if (userRef.equals(ACCESS_HELPER.getOwner(this.patient).getUser())) {
             return true;
         }
+        Set<Group> userGroups = GROUP_MANAGER.getGroupsForUser(currentUser);
+        for (Group group : userGroups) {
+            DocumentReference groupRef = group.getReference();
+            if (groupRef.equals(ACCESS_HELPER.getOwner(this.patient).getUser())) {
```

Comment:
i think it makes sense to cache `ACCESS_HELPER.getOwner(this.patient).getUser()` in a variable